### PR TITLE
initial pants configuration basic structure

### PR DESCRIPTION
**Pull request by TonySherman**
In an attempt to keep PRs small, I've added some of the very basic Pants configuration and just basic folder structure for a possible example microservice.

### CODE REVIEW

File: requirements.txt
Comment (huonw) [inline requirements.txt:1]: It looks like this includes all transitive dependencies pinned. I think for this example repo, it'd be good to just include the top-level ones with version ranges (or fully un-pinned) and rely on Pants' lockfile, keeping it "simpler".

Potentially we could include them in "new-style" `pyproject.toml` `[dependencies]`?

**Review by huonw — APPROVED** [review]
Thank you very much for proposing this idea and starting on it. I like the broad structure of what you have so far.

I know "microservices" and some architectures involving them (like Lambda-per-HTTP-endpoint) can be controversial, so I'd want to try to hint that Pants is agnostic to exactly what Lambdas are being used for. For instance, Pants works equally well to build lambdas for Lambda@Edge or SNS/SQS subscribers so it'd be nifty to include one or two of them, eventually. What do you think?

Along those lines, maybe the `awslambda_example` top-level package could be renamed to indicate (conceptually) what's in there? E.g. `api_lambdas` or something?

Finally, since this is an example repository, I think it'd be good to have it be easy to understand, and I'm a bit nervous about having too many separate `BUILD` files for that. What do you think about using `awslambda_example/create_user.py` (etc.), without the extra folder layer?

**Comment by TonySherman** [comment]
> I know "microservices" and some architectures involving them (like Lambda-per-HTTP-endpoint) can be controversial, so I'd want to try to hint that Pants is agnostic to exactly what Lambdas are being used for. For instance, Pants works equally well to build lambdas for Lambda@Edge or SNS/SQS subscribers so it'd be nifty to include one or two of them, eventually. What do you think?

I'm totally on board with that! I originally started with the lambda per endpoint because I feel like Pants is one of the best tools to help package code for that use case. The fact that you can share code across several lambdas and Pants will package properly is amazing.

I recently heard a talk that proposed one lambda for create/update/delete and a separate for read only. It was kind of a compromise between the micro model and the lambda-lith. Would you like the example to demonstrate that? It would certainly show both use cases.

> Along those lines, maybe the `awslambda_example` top-level package could be renamed to indicate (conceptually) what's in there? E.g. `api_lambdas` or something?

Yeah. I'll make that change. I was originally thinking there could also be gcp examples in this repo so that's how I came up with that.

> Finally, since this is an example repository, I think it'd be good to have it be easy to understand, and I'm a bit nervous about having too many separate `BUILD` files for that. What do you think about using `awslambda_example/create_user.py` (etc.), without the extra folder layer?

I will make this change as well. I became so used to using a single BUILD file for each lambda that I didn't even think of this.


Thanks for the review and feedback! I'm going to work on the suggested changes and get them pushed up.

**Comment by huonw** [comment]
> I recently heard a talk that proposed one lambda for create/update/delete and a separate for read only. It was kind of a compromise between the micro model and the lambda-lith. Would you like the example to demonstrate that? It would certainly show both use cases.

I'm happy enough with lambda per endpoint (as you have it) to demonstrate building an HTTP API. I think just one or two that respond to other events would be cool too. 👍 

> Yeah. I'll make that change. I was originally thinking there could also be gcp examples in this repo so that's how I came up with that.

Ah, yeah, I guess we could have a `aws/...` directory and `gcp/...` directory?

File: requirements.txt
Comment (TonySherman) [reply requirements.txt:1]: I haven't used pyproject.toml for dependencies yet so I switched it to the example from the pants docs. Also, not even using the only dependency yet but planning to in my example code.

**Comment by TonySherman** [comment]
Let me know what you think of this. Once this is merged in, I can get a PR up with some example code. I'll start with the API but going to keep in mind some of the other lambda use cases too. Thanks @huonw!

**Review by huonw — APPROVED** [review]

**Comment by huonw** [comment]
Thank you, and thanks for waiting!

**Pull request merged**